TimeOfFlight/src/TOFEstimators.cc

```diff
@@ -33,442 +43,593 @@
 
+        _h[2] = new TH2F( "hbetaFirstHitsNeut", "beta vs momentum - first hit - neutral",    nBins, .1 , 10., nBins, 0.93 , 1.03 ) ;
+        _h[3] = new TH2F( "hbetaCloseHitsNeut", "beta vs momentum - closest hits - neutral", nBins, .1 , 10., nBins, 0.93 , 1.03 ) ;
 
-  // create some histograms with beta vs momentum for charged particles
-  
-  if( isFirstEvent() ){
+        _h[4] = new TH2F( "hbetaLastTrkHit",    "beta vs momentum - last tracker hit", nBins, .1 , 10., nBins, 0.93 , 1.03 ) ;
 
-    // this creates a directory for this processor ....
-    AIDAProcessor::histogramFactory( this ) ;
 
-    _h.resize(5) ;
-    int nBins = 100 ;
-    _h[0] = new TH2F( "hbetaFirstHitsChrg", "beta vs momentum - first hit - charged",    nBins, .1 , 10., nBins, 0.93 , 1.03 ) ; 
-    _h[1] = new TH2F( "hbetaCloseHitsChrg", "beta vs momentum - closest hits - charged", nBins, .1 , 10., nBins, 0.93 , 1.03 ) ; 
+      }
+
+      // get the PFO collection from the event if it exists
+      LCCollection* colPFO = nullptr ;
+      try{ colPFO = evt->getCollection( _colNamePFO ) ; } catch(lcio::Exception&){}
+
+      if( colPFO != nullptr ){
+
+        PIDHandler pidh( colPFO );
+        int algoID       = pidh.getAlgorithmID( name() );
+        int tof_firsthit = pidh.getParameterIndex(algoID,"TOFFirstHit") ;
+        int tof_closest  = pidh.getParameterIndex(algoID,"TOFClosestHits") ;
+        int tof_length   = pidh.getParameterIndex(algoID,"TOFFlightLength") ;
+        int tof_trkhit   = pidh.getParameterIndex(algoID,"TOFLastTrkHit") ;
+        int tof_trk_len  = pidh.getParameterIndex(algoID,"TOFLastTrkHitFlightLength") ;
+
+        int nPFO = colPFO->getNumberOfElements()  ;
+
+        for( int i=0 ; i< nPFO ; ++i){
+
+          ReconstructedParticle* pfo = dynamic_cast<ReconstructedParticle*>(  colPFO->getElementAt( i ) ) ;
+
+          const ParticleID& tofPID = pidh.getParticleID( pfo , algoID ) ;
+
+          const FloatVec& tofParams = tofPID.getParameters() ;
 
-    _h[2] = new TH2F( "hbetaFirstHitsNeut", "beta vs momentum - first hit - neutral",    nBins, .1 , 10., nBins, 0.93 , 1.03 ) ; 
-    _h[3] = new TH2F( "hbetaCloseHitsNeut", "beta vs momentum - closest hits - neutral", nBins, .1 , 10., nBins, 0.93 , 1.03 ) ; 
+          streamlog_out( DEBUG ) << " ****  found TOF parameters for pfo w/ size " << tofParams.size() << std::endl ;
 
-    _h[4] = new TH2F( "hbetaLastTrkHit",    "beta vs momentum - last tracker hit", nBins, .1 , 10., nBins, 0.93 , 1.03 ) ; 
+          if( !tofParams.empty() ){
 
+    	const double* mom = pfo->getMomentum() ;
+    	double momentum = sqrt( mom[0] * mom[0] +  mom[1] * mom[1] +  mom[2] * mom[2] ) ;
 
-  }
+    	double length  =  tofParams[ tof_length  ] ;
 
-  // get the PFO collection from the event if it exists
-  LCCollection* colPFO = nullptr ;
-  try{ colPFO = evt->getCollection( _colNamePFO ) ; } catch(lcio::Exception){}
+    	double beta_fh = ( length / tofParams[ tof_firsthit] ) / 299.8 ;
```

Comment:
Why to use a truncated 299.8 when have the constant available from CLHEP. Please it change everywhere

Reply:
I know this is not from your algorithms, but if you want to make comparison studies, better not to be biased from this approximation

Reply:
It is 2018 Frank's part of the code and changing that will break backwards compatability

Reply:
You was faster :)
Ok ,will change